src/main/java/com/fasterxml/jackson/databind/deser/std/StdDeserializer.java

```diff
@@ -1421,13 +1421,16 @@ protected final String _parseString(JsonParser p, DeserializationContext ctxt,
             return ob.toString();
         // 29-Jun-2020, tatu: New! "Scalar from Object" (mostly for XML)
         case JsonTokenId.ID_START_OBJECT:
+            act = _checkObjectToStringCoercion(p, ctxt, rawTargetType);
```

Comment:
Hmmmh. Let's not add this call -- the thing is that this only applies to XML backend, and is more of a work-around in that case, not a real coercion. For all other backends, including JSON, there will be failure for incompatible coercion.

So for time being, let's remove this check.